backend/src/routes/post.ts

```diff
@@ -58,7 +59,17 @@ router.post('/:postID/unlike', async (ctx, next) => {
 	await post.save();
 	ctx.body = post;
 });
-
+router.post('/:postID/save', async (ctx, next) => {
+	const post = await Post.findOneBy(mongodb.ObjectId(ctx.params.postID));
+	if (post === null) {
+		ctx.status = 404;
+		return;
+	}
+	const user = ctx.state.user.id;
+	user.savedPosts.push(post.id);
```

Comment:
ctx.state.user.id is just their id, so you'll have to use findOneBy with this id to grab the model first